### PR TITLE
Show numpy and scipy versions in `show_config`

### PR DESCRIPTION
**Pull request by emcastillo**
Currently `cupy.show_config` only shows cuda related info.
In order to easily lookup for errors in CI and locally, printing numpy and scipy versions as well can be helpful.

Blocked by #3858 and #3877

### CODE REVIEW

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
This closes #3629.

**Comment by emcastillo** [comment]
TBD, show os, cc of the device and all the relevant info we can gather

**Comment by emcastillo** [comment]
rebased

```
OS                           : Linux-4.15.0-112-generic-x86_64-with-debian-buster-sid
CuPy Version                 : 8.0.0rc1
NumPy Version                : 1.18.5
SciPy Version                : 1.4.1
CUDA Root                    : /usr/local/cuda-10.2
CUDA Build Version           : 11000
CUDA Driver Version          : 11000
CUDA Runtime Version         : 11000
cuBLAS Version               : 11200
cuFFT Version                : 10201
cuRAND Version               : 10201
cuSOLVER Version             : (10, 6, 0)
cuSPARSE Version             : 11101
NVRTC Version                : (11, 0)
Thrust Version               : 100909
CUB Build Version            : 100909
cuDNN Build Version          : 7500
cuDNN Version                : 7500
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
Device 0 Name                : Tesla V100-PCIE-16GB
Device 0 Compute Capability  : 70
```

**Review by kmaehashi — APPROVED** [review]
LGTM!

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: cupyx/_runtime.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import inspect
 import io
 import os
+import platform
 
 import cupy
+import numpy
```
Comment (toslunar) [inline cupyx/_runtime.py:7]:
```suggestion
import numpy

import cupy
```

**Review by toslunar — CHANGES_REQUESTED** [review]

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**